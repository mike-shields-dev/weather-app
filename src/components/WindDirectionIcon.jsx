import React from "react";
import PropTypes from "prop-types";
import directionLookup from "../data/directionLookup.json";
import styles from "../styles/WindDirectionIcon.module.css";

const WindDirectionIcon = ({ direction }) => {
  return (
    <svg
      className={styles["wind-direction-icon"]}
      data-testid="wind-direction-icon"
      version="1.1"
      x="0px"
      y="0px"
      viewBox="0 0 490 490"
      fill="currentColor"
      style={{
        transform: `rotate(${directionLookup[direction].degrees}deg)`,
      }}
    >
      <title>{`arrow pointing ${directionLookup[direction].humanReadable}`}</title>
      <path d="M480.086,490L245,339.229L9.914,490L245,0L480.086,490z" />
    </svg>
  );
};

WindDirectionIcon.propTypes = {
  direction: PropTypes.string.isRequired,
};

export default WindDirectionIcon;
