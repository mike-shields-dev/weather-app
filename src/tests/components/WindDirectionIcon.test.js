import React from "react";
import { render, screen } from "@testing-library/react";
import WindDirectionIcon from "../../components/WindDirectionIcon";
import {
  degreesFromWindDirection,
  unabbreviatedWindDirection,
} from "../../helpers/windDirection";

const windDirections = [
  "n",
  "nne",
  "ne",
  "ene",
  "e",
  "ese",
  "se",
  "sse",
  "s",
  "ssw",
  "sw",
  "wsw",
  "w",
  "wnw",
  "nw",
  "nnw",
];

describe("WindDirectionIcon", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<WindDirectionIcon direction="n" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders an icon pointing in the provided direction", async () => {
    windDirections.forEach(async (direction) => {
      render(<WindDirectionIcon direction={direction} />);

      const WindDirectionIconComponent = await screen.findByTestId(
        "wind-direction-icon"
      );

      expect(WindDirectionIconComponent).toHaveStyle(
        `transform: rotate(${degreesFromWindDirection(direction)}deg)`
      );
    });
  });

  it("renders an svg with a title describing the direction", () => {
    windDirections.forEach(async (direction) => {
      render(<WindDirectionIcon direction={direction} />);

      const descriptionRegex = new RegExp(
        unabbreviatedWindDirection(direction),
        "i"
      );

      const windDirectionDescription = await screen.findByText(
        descriptionRegex
      );

      expect(windDirectionDescription).toBeInTheDocument();
    });
  });
});
