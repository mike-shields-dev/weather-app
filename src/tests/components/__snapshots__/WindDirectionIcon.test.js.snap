// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindDirectionIcon matches snapshot 1`] = `
<DocumentFragment>
  <svg
    class="wind-direction-icon"
    data-testid="wind-direction-icon"
    fill="currentColor"
    style="transform: rotate(0deg);"
    version="1.1"
    viewBox="0 0 490 490"
    x="0px"
    y="0px"
  >
    <title>
      arrow pointing north
    </title>
    <path
      d="M480.086,490L245,339.229L9.914,490L245,0L480.086,490z"
    />
  </svg>
</DocumentFragment>
`;
